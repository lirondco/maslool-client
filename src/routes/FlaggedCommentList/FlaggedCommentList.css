.flagged_comments {
  display: flex;
  flex-direction: column;
}

.FlaggedCommentList {
  margin: 2%;
  padding: 2%;
}

.FlaggedCommentList h2 {
  margin-bottom: 1%;
}

hr {
  border: 1px solid;
  margin-bottom: 4%;
}
