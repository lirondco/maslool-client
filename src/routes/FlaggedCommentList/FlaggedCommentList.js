import React, { Component } from 'react'
import ModeratorApiService from '../../services/moderator-api-service'
import FlaggedComment from '../../components/FlaggedComment/FlaggedComment'
import './FlaggedCommentList.css'

export default class FlaggedCommentList extends Component {
    state =  {
        error: null,
        comments: null,
    }

    getFlaggedComments = () => {
        ModeratorApiService.getFlaggedComments()
            .then(comments => this.setState({ comments }))
            .then(() => this.setState({error: null}))
            .catch(error => this.setState({ ...error }))
    }

    componentDidMount = () => {
        this.getFlaggedComments()
    }

    handleBanUser = (userId) => {
        ModeratorApiService.toggleBanUser(userId)
            .then(() => this.getFlaggedComments())
            .then(() => window.alert('Do not forget to edit and/or unflag their comment!'))
            .catch(error => this.setState({ ...error }))
    }

    renderComments = (comments) => {
        return (
            <ul className='flagged_comments'>
                {comments.map(comment =>
                    <li key={comment.id}>
                        <FlaggedComment comment={comment} onBanClick={() => this.handleBanUser(comment.user.id)} />
                    </li>
                    )}
            </ul>
        )
    }
    
    render() {
        const { comments, error } = this.state

        return (
            <section className='FlaggedCommentList'>
                <h2>Flagged comments</h2>
                <hr />
                {comments && this.renderComments(comments)}
                {comments === null && <h4>List is empty</h4>}
                <div role="alert">{error && <p className="error">{error}</p>}</div>
            </section>
        )
    }

}