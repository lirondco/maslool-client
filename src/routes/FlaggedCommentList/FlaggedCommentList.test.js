import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FlaggedCommentList from "./FlaggedCommentList";

describe(`FlaggedCommentList component`, () => {
  it("renders the complete list", () => {
    const wrapper = shallow(<FlaggedCommentList />);
    wrapper.setState({
      error: null,
      comments: [
        {
          id: 2,
          content:
            "Mister Krabs is lying do not believe him he has not paid me my last salary!!!",
          last_modified: "2021-03-11T06:47:06.477Z",
          flagged: true,
          flagged_by: "squidward",
          trail_id: 3,
          user: {
            id: 1,
            username: "spongebob",
            join_date: "2021-03-11T06:47:06.477468",
            admin: false,
            banned: false,
          },
        },
        {
          id: 9,
          content: "why did my friend get banned? ",
          last_modified: "2021-03-20T17:37:56.565Z",
          flagged: true,
          flagged_by: "squidward",
          trail_id: 1,
          user: {
            id: 1,
            username: "spongebob",
            join_date: "2021-03-11T06:47:06.477468",
            admin: false,
            banned: false,
          },
        },
      ],
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
