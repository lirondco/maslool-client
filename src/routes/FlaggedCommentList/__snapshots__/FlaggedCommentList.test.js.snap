// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlaggedCommentList component renders the complete list 1`] = `
<section
  className="FlaggedCommentList"
>
  <h2>
    Flagged comments
  </h2>
  <hr />
  <ul
    className="flagged_comments"
  >
    <li
      key="2"
    >
      <FlaggedComment
        comment={
          Object {
            "content": "Mister Krabs is lying do not believe him he has not paid me my last salary!!!",
            "flagged": true,
            "flagged_by": "squidward",
            "id": 2,
            "last_modified": "2021-03-11T06:47:06.477Z",
            "trail_id": 3,
            "user": Object {
              "admin": false,
              "banned": false,
              "id": 1,
              "join_date": "2021-03-11T06:47:06.477468",
              "username": "spongebob",
            },
          }
        }
        onBanClick={[Function]}
      />
    </li>
    <li
      key="9"
    >
      <FlaggedComment
        comment={
          Object {
            "content": "why did my friend get banned? ",
            "flagged": true,
            "flagged_by": "squidward",
            "id": 9,
            "last_modified": "2021-03-20T17:37:56.565Z",
            "trail_id": 1,
            "user": Object {
              "admin": false,
              "banned": false,
              "id": 1,
              "join_date": "2021-03-11T06:47:06.477468",
              "username": "spongebob",
            },
          }
        }
        onBanClick={[Function]}
      />
    </li>
  </ul>
  <div
    role="alert"
  />
</section>
`;
